subsys/nrf_security/src/drivers/cracen/cracenpsa/src/common.c

```diff
@@ -125,27 +125,27 @@ static psa_status_t get_sx_brainpool_curve(size_t curve_bits, const struct sx_pk
 
 	switch (curve_bits) {
 	case 192:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP192r1));
 		break;
 	case 224:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP224r1));
 		break;
 	case 256:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1_256,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP256r1));
 		break;
 	case 320:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP320r1));
 		break;
 	case 384:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1_384,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP384r1));
 		break;
 	case 512:
-		IF_ENABLED(PSA_NEED_CRACEN_KEY_MANAGEMENT_BRAINPOOL_P_R1_512,
+		IF_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY,
 			   (selected_curve = &sx_curve_brainpoolP512r1));
```

Comment:
won't this new system put all or none of the curves in the binary whereas before we were able to just put the curves we wanted into the binary?

Reply:
I compared the map file of HCS and NCS.

This new config system pulls in all the curves whereas the old did not.

For the ecjpake sample we are consuming an extra 11kB of FLASH.

old:
           FLASH:       63000 B      1524 KB      4.04%
             RAM:       32912 B       256 KB     12.55%


new:
           FLASH:       74032 B      1524 KB      4.74%
             RAM:       24696 B       256 KB      9.42%

Reply:
Yes it does that is due to new PSA Core that is more detailed on the Key types than the old one we have follow up task (NCSDK-26237) tho add a config that separates the curves, it is planned for the next sprint after this gets in

Reply:
ah, I was not aware of this follow-up task. Good to know!